src/model_service.cpp

```diff
@@ -73,16 +84,27 @@ Status GetModelStatusImpl::serializeResponse2Json(const tensorflow::serving::Get
     return StatusCode::OK;
 }
 
-Status GetModelStatusImpl::getModelStatus(const tensorflow::serving::GetModelStatusRequest* request, tensorflow::serving::GetModelStatusResponse* response) {
+Status GetModelStatusImpl::getModelStatus(
+    const tensorflow::serving::GetModelStatusRequest* request,
+    tensorflow::serving::GetModelStatusResponse* response,
+    ModelManager& manager) {
     SPDLOG_DEBUG("model_service: request: {}", request->DebugString());
 
     bool has_requested_version = request->model_spec().has_version();
     auto requested_version = request->model_spec().version().value();
     std::string requested_model_name = request->model_spec().name();
-    auto model_ptr = ModelManager::getInstance().findModelByName(requested_model_name);
+    auto model_ptr = manager.findModelByName(requested_model_name);
     if (!model_ptr) {
-        SPDLOG_WARN("requested model {} was not found", requested_model_name);
-        return StatusCode::MODEL_NAME_MISSING;
+        SPDLOG_DEBUG("GetModelStatus: Model {} is missing, trying to find pipeline with such name", requested_model_name);
+        auto pipelineDefinition = manager.getPipelineFactory().findDefinitionByName(requested_model_name);
+        if (!pipelineDefinition) {
+            return StatusCode::MODEL_NAME_MISSING;
```

Comment:
What if user requests version=2?

Reply:
Any version requested will refer to version 1. The same logic goes for predict and getmodelmetadata.  Since we do not version pipeline definitions in any way we allow requesting any version.

Reply:
Added test for version=5